client/src/main/scripts/components/search/searchUtils.ts

```diff
@@ -80,12 +80,18 @@ export class TurnFinder {
       }
     }
     if (this.regexExclude) {
-      // TODO: in a second pass, if this exclude regex matches at all,
-      // then we should not remove that found entry
-      //
-      // it's a little tricky because it might be a "look ahead" or "look behind", e.g.
-      //   wall, -wall street <-- I want wall, but not wall street
-      //   wall, -border wall <-- I want wall, but not border wall
+      this.regexExclude.lastIndex = 0;
+      let foundNegative = this.regexExclude.exec(turnContent);
+      while (foundNegative != null) {
+        const foundNegativeHunk = foundNegative[0];
+        const lastNegativeIndex =
+          foundNegative.index + foundNegativeHunk.length;
+        foundOffsets = foundOffsets.filter((fo) => {
+          return fo[0] !== foundNegative?.index;
```

Comment:
I think this works for `wall, -wall street`, but not for `wall, -border wall`, and especially not for `states, -united states of america`